crates/attestation/src/report_data.rs

```diff
@@ -34,7 +34,8 @@ impl ReportDataVersion {
 
 #[derive(Debug, Clone, Constructor)]
 pub struct ReportDataV1 {
-    tls_public_key: near_sdk::PublicKey,
+    // This corresponds to an Ed25519 public key
+    tls_public_key: [u8; 32],
```

Comment:
I’d prefer to have a self-explanatory strong type here rather than a raw byte array with a comment explaining it, tbh.

Reply:
One option is to have a named bytes wrapper defined in the attestation crate. It would convey that this crate only cares about the raw bytes while still being a self-explanatory strong type that doesn't need a comment. What do you think of that?

Something like:
```
struct Ed25519PubKey([u8; 32]);
```

and deriving all appropriate traits for it (i.e. Clone, Copy, Hash, Deref, AsRef, Borrow etc.)

Reply:
ok, fixed in 04157008ae78bf844d6442caf3e4f71996d92cd1

Reply:
> by Borrow you mean std::borrow::Borrow, it seems I cannot use std here?

Yes but this one should be in `core` as well as `core::borrow::Borrow`.

Reply:
> It also prevents us from accidentally passing another [u8; 32] value here (like a hash output or whatever).

How does it prevent that? The current solution allows the input of [u8; 32], else we would need to make this new type public and use conversion every time it is used? Was that the point?

Reply:
Right yeah that's correct. Although putting the newtype in the interface makes the expected type clearer, and then the caller could choose whether or not they want to convert on their end.